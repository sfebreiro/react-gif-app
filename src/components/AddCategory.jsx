import { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('')

    const onInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newCategory = inputValue.trim();
        
        if (newCategory.length <= 1) return;

        setInputValue('');
        
        //setCategories(categories => [ inputValue, ...categories ]);
        onNewCategory(newCategory);

    }


  return (
    <form onSubmit={ onSubmit } aria-label='form' >

        <input 
            type="text" 
            placeholder="Buscar gifs" 
            value={ inputValue } 
            onChange={ onInputChange }
            
        />

    </form>

    
  )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}