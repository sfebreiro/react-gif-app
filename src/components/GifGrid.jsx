import React, { Fragment, useEffect, useState } from 'react';
import { PropTypes } from 'prop-types';

import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );


  return (
    <Fragment>
        <h3> { category } </h3>

        {
            isLoading && (<h4>Cargando...</h4>)
        }

        <div className='card-grid'>
            {
                images.map( ( image ) => (
                    <GifItem 
                        key={ image.id } 
                        // title={ image.title }
                        // url={ image.url }
                        { ...image }
                    />
                ))
            }
        </div>
    </Fragment>
  )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}