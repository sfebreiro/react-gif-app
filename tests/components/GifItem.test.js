import { render, screen } from '@testing-library/react';
import { GifItem } from "../../src/components/GifItem";


describe('first', () => { 

    const title = 'Titulo'
    const url = 'https://urlstring.com/'
    
    test('Debe de hacer match con el snapshot', () => { 
        
        const { container } = render( <GifItem title={ title } url={ url } /> );

        expect(container).toMatchSnapshot();

     }) 


     test('Debe de mostrar la imagen con la URL indicada', () => { 
        
        render( <GifItem title={ title } url={ url } /> );
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(url);

        const { alt, src } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

      })


      test('Debe de mostrar el tÃ­tulo en el componente', () => { 
        
        render( <GifItem title={ title } url={ url } /> );
        expect(screen.getByText(title)).toBeTruthy();

       })

})