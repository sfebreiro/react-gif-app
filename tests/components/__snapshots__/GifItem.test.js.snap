// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://urlstring.com/"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
